### PR TITLE
Fix statement type coverage calculation

### LINKED ISSUE #325: Coverage calculation results have changed since v0.57.0
When using v0.57.0, gocover's coverage calculation results are lower than v0.56.x.
For example, v0.56.4 is used for test data from this repository, the coverage would be as follows:

```
~/ghq/github.com/k1LoW/octocov/coverage/testdata/gocover main
❯ octocov -v
octocov version 0.56.4

~/ghq/github.com/k1LoW/octocov/coverage/testdata/gocover main
❯ octocov -r coverage.out

            main (faa6ab0)
----------------------------
  Coverage           68.0%
```

But with v0.57.0, the coverage would be as follows:

```
~/ghq/github.com/k1LoW/octocov/coverage/testdata/gocover main
❯ octocov -v
octocov version 0.57.0

~/ghq/github.com/k1LoW/octocov/coverage/testdata/gocover main
❯ octocov -r coverage.out

            main (faa6ab0)
----------------------------
  Coverage           65.1%
```

I thin the problem is caused by the `(*Coverage).reCalc()` method being called.
https://github.com/k1LoW/octocov/blob/faa6ab086a685c2d33531f5933929b8ece7fbd47/coverage/merge.go#L25-L39

This method calculates file coverage from line coverage. However, this approach is not appropriate because the coverage representation of gocover is different from other formats.

For example, in the following block, the value of "numberOfStatements" is 1 for lines 45 through 47, so the number of rows to be recorded is 1, not 3.

https://github.com/k1LoW/octocov/blob/faa6ab086a685c2d33531f5933929b8ece7fbd47/coverage/testdata/gocover/coverage.out#L12

However, if the `(BlockCoverages).ToLineCoverages()` method is called, the line coverage for this block is counted as 3.

### PR DESCRIPTION
**Pull request by furusax0621**
Fix #325

### CODE REVIEW

**Comment by k1LoW** [comment]
@furusax0621 
Fixed govulncheck in main branch

**Comment by furusax0621** [comment]
API rate limit 😢 
https://github.com/k1LoW/octocov/actions/runs/7898241942/job/21555315932?pr=327#step:6:101

**Comment by k1LoW** [comment]
Please wait a moment 🙏 

```
 gh_test.go:79: GET https://api.github.com/repos/k1LoW/octocov: 403 API rate limit exceeded for installation ID 1750184. If you reach out to GitHub Support for help, please include the request ID B8A9:92DD:BF354D:17F4B38:65CC767D. [rate reset in 21m11s]
```

File: coverage/merge.go [hunk 1]
```diff
@@ -26,11 +29,27 @@ func (c *Coverage) reCalc() error {
...
 	covered := 0
 	for _, f := range c.Files {
-		lcs := f.Blocks.ToLineCoverages()
-		f.Total = lcs.Total()
-		f.Covered = lcs.Covered()
-		total += f.Total
-		covered += f.Covered
+		var fileTotal, fileCovered int
+
+		switch f.Type {
```
Comment (k1LoW) [inline coverage/merge.go:34]: Thank you!!

**Review by k1LoW — APPROVED** [review]
GREAT! Thank you!!

**Pull request merged**